src/Beman/Optional26/tests/detail/iterator.t.cpp

```diff
@@ -0,0 +1,178 @@
+/**
+ * This file contains tests for beman::optional::detail::normal_iterator.
+ */
```

Comment:
Why do we have a tests/detail here -- shouldn't it just be tests?  I don't think we need to 'hide' some sort of implementation details for test code.

Reply:
If we're not putting the tests in the directory with the code, putting them in a mechanically derivable location makes it easier / possible to teach an IDE or editor to open related files. 
This is a lot of ceremony that seems unneeded on a project with one real file, but with larger projects knowing where things are and go makes it much easier to navigate.

Reply:
The library is currently building with std 23. I think it would probably work with 20. Deducing this is the "killer feature" for  libraries in 23, but it's not used here. 

Writing in 20 style makes translation to and from standard wording *very* straightforward.

Reply:
I'm saying this test file (and all) should be in Optional26/tests -- NOT Optional26/tests/detail. I'm not following why we need that extra directory level and then basically leaving tests empty?

Reply:
I prefer to have a tree identical for production code and tests code.

Reply:
Ok well I consider it strange. What I was missing is that already on main is the public interface tests.

Reply:
yes, I merged it in a previous PR with initial implementation using T*

Reply:
I propose to move this discussion into a Beman Standard thread. To be honest, I'm happy to change it to any convention if we have a guideline.